WalletWasabi.Fluent/ViewModels/Wallets/Send/LabelSelectionViewModel.cs

```diff
@@ -15,45 +15,48 @@ public partial class LabelSelectionViewModel : ViewModelBase
 
 	[AutoNotify] private bool _enoughSelected;
 
-	private Pocket? _privatePocket;
+	private Pocket _privatePocket = Pocket.Empty;
 	private bool _includePrivatePocket;
+	private Pocket[] _allPockets = Array.Empty<Pocket>();
 
 	public LabelSelectionViewModel(Money targetAmount)
 	{
 		_targetAmount = targetAmount;
 	}
 
-	public Pocket[] AllPockets { get; set; } = Array.Empty<Pocket>();
+	public Pocket[] NonPrivatePockets { get; set; } = Array.Empty<Pocket>();
 
 	public IEnumerable<LabelViewModel> AllLabelsViewModel { get; set; } = Array.Empty<LabelViewModel>();
 
 	public IEnumerable<LabelViewModel> LabelsWhiteList => AllLabelsViewModel.Where(x => !x.IsBlackListed);
 
 	public IEnumerable<LabelViewModel> LabelsBlackList => AllLabelsViewModel.Where(x => x.IsBlackListed);
 
-	public Pocket[] GetSafeToUsePockets()
+	public Pocket[] AutoSelectPockets()
 	{
-		var pockets = AllPockets.ToList();
+		var knownPockets = NonPrivatePockets.Where(x => x.Labels != CoinPocketHelper.UnlabelledFundsText).ToArray();
 
-		if (_privatePocket is { } privatePocket)
+		if (_privatePocket.Amount >= _targetAmount)
```

Comment:
I think once we move the recipient label to the initial send page we can add another check before this one to select the pocket that is only known by the recipient and not even use the private pocket.

Reply:
Yes, it can be added once the recipient dialog is in the correct place.